File: src/main/java/com/github/lizhanyin/tfs/wizard/step/TeamProjectSelectionStep.java
package com.github.lizhanyin.tfs.wizard.step;

import com.github.lizhanyin.tfs.services.TfsConnectionService;
import com.github.lizhanyin.tfs.wizard.ImportProjectContext;
import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.progress.ProgressIndicator;
import com.intellij.openapi.progress.ProgressManager;
import com.intellij.openapi.progress.Task;
import com.intellij.ui.CollectionComboBoxModel;
import com.intellij.ui.components.JBLabel;
import com.intellij.util.ui.FormBuilder;
import com.intellij.util.ui.JBUI;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.awt.*;
import java.util.ArrayList;
import java.util.List;

/**
 * 团队项目选择步骤
 */
public class TeamProjectSelectionStep extends AbstractWizardStep {

    private static final String STEP_ID = "team-project-selection";

    // UI 组件
    private JComboBox<String> teamProjectComboBox;
    private JBLabel statusLabel;
    private JButton refreshButton;
    private JLabel serverInfoLabel;

    // 团队项目列表
    private final List<String> teamProjects = new ArrayList<>();

    public TeamProjectSelectionStep(@NotNull ImportProjectContext context) {
        super(STEP_ID, "选择团队项目", context);
    }

    @Override
    protected JComponent buildComponent() {
        FormBuilder builder = FormBuilder.createFormBuilder();

        // 服务器信息
        serverInfoLabel = new JLabel(" ");
        updateServerInfo();
        builder.addLabeledComponent("已连接到:", serverInfoLabel);
        builder.addSeparator();
        builder.addLabeledComponent("团队项目:", createTeamProjectPanel());
        builder.addSeparator();
        builder.addComponent(createStatusPanel());

        JPanel panel = builder.getPanel();
        panel.setBorder(JBUI.Borders.empty(10));

        return panel;
    }

    /**
     * 更新服务器信息显示
     */
    private void updateServerInfo() {
        StringBuilder info = new StringBuilder("<html><b>");
        info.append(context.getServerUrl());
        if (context.getCollectionName() != null && !context.getCollectionName().isEmpty()) {
            info.append("/").append(context.getCollectionName());
        }
        info.append("</b></html>");
        serverInfoLabel.setText(info.toString());
    }

    /**
     * 创建团队项目选择面板
     */
    private JComponent createTeamProjectPanel() {
        JPanel panel = new JPanel(new BorderLayout(5, 0));

        teamProjectComboBox = new JComboBox<>();
        teamProjectComboBox.setEnabled(false);
        updateTeamProjectComboBox();
        panel.add(teamProjectComboBox, BorderLayout.CENTER);

        JPanel buttonPanel = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 0));

        refreshButton = new JButton("刷新");
        refreshButton.addActionListener(e -> loadTeamProjects());
        buttonPanel.add(refreshButton);

        panel.add(buttonPanel, BorderLayout.EAST);

        return panel;
    }

    /**
     * 创建状态面板
     */
    private JComponent createStatusPanel() {
        JPanel panel = new JPanel(new BorderLayout());
        statusLabel = new JBLabel("点击\"刷新\"加载团队项目列表");
        panel.add(statusLabel, BorderLayout.CENTER);
        return panel;
    }

    /**
     * 加载团队项目列表
     */
    private void loadTeamProjects() {
        if (!context.isServerConfigured()) {
            statusLabel.setText("请先配置服务器连接");
            statusLabel.setForeground(java.awt.Color.RED);
            return;
        }

        statusLabel.setText("正在加载团队项目...");
        statusLabel.setForeground(java.awt.Color.BLACK);
        refreshButton.setEnabled(false);
        teamProjectComboBox.setEnabled(false);

        ProgressManager.getInstance().run(new Task.Backgroundable(null, "加载团队项目", true) {
            @Override
            public void run(@NotNull ProgressIndicator indicator) {
                indicator.setText("正在连接到 TFS 服务器...");
                indicator.setIndeterminate(true);

                try {
                    // 使用 TFS SDK 加载团队项目列表
                    TfsConnectionService connectionService =
                            ApplicationManager.getApplication().getService(TfsConnectionService.class);

                    List<String> projects = connectionService.getTeamProjects(context);

                    // 更新 UI
                    SwingUtilities.invokeLater(() -> {
                        teamProjects.clear();
                        teamProjects.addAll(projects);
                        updateTeamProjectComboBox();

                        if (projects.isEmpty()) {
                            statusLabel.setText("未找到团队项目");
                            statusLabel.setForeground(java.awt.Color.ORANGE);
                        } else {
                            statusLabel.setText("已加载 " + projects.size() + " 个团队项目");
                            statusLabel.setForeground(new java.awt.Color(0, 128, 0));
                        }

                        refreshButton.setEnabled(true);
                        teamProjectComboBox.setEnabled(true);
                    });

                } catch (Exception e) {
                    SwingUtilities.invokeLater(() -> {
                        statusLabel.setText("加载失败: " + e.getMessage());
                        statusLabel.setForeground(java.awt.Color.RED);
                        refreshButton.setEnabled(true);
                    });
                }
            }
        });
    }

    /**
     * 更新团队项目下拉框
     */
    private void updateTeamProjectComboBox() {
        teamProjectComboBox.setModel(new CollectionComboBoxModel<>(teamProjects));
        if (!teamProjects.isEmpty()) {
            teamProjectComboBox.setSelectedIndex(0);
        }
    }

    @Override
    public boolean isComplete() {
        return teamProjectComboBox.getSelectedItem() != null;
    }

    @Override
    public JComponent getPreferredFocusedComponent() {
        return teamProjectComboBox;
    }

    @Override
    public boolean onFinish() {
        String selected = (String) teamProjectComboBox.getSelectedItem();
        if (selected != null) {
            context.setTeamProject(selected);
        }
        return true;
    }
}

File: src/main/kotlin/com/github/lizhanyin/tfs/wizard/step/TeamProjectSelectionStep.kt
package com.github.lizhanyin.tfs.wizard.step

import com.github.lizhanyin.tfs.TfsBundle
import com.github.lizhanyin.tfs.services.TfsConnectionService
import com.github.lizhanyin.tfs.wizard.ImportProjectContext
import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.progress.ProgressIndicator
import com.intellij.openapi.progress.ProgressManager
import com.intellij.openapi.progress.Task
import com.intellij.ui.CollectionComboBoxModel
import com.intellij.ui.components.JBLabel
import com.intellij.util.ui.FormBuilder
import com.intellij.util.ui.JBUI
import java.awt.BorderLayout
import java.awt.Color
import java.awt.FlowLayout
import javax.swing.JButton
import javax.swing.JComboBox
import javax.swing.JComponent
import javax.swing.JLabel
import javax.swing.JPanel
import javax.swing.SwingUtilities

/**
 * 团队项目选择步骤
 */
class TeamProjectSelectionStep(context: ImportProjectContext) :
    AbstractWizardStep(STEP_ID, TfsBundle.message("wizard.step.teamProject.title"), context) {

    companion object {
        private const val STEP_ID = "team-project-selection"
    }

    private lateinit var teamProjectComboBox: JComboBox<String>
    private lateinit var statusLabel: JBLabel
    private lateinit var refreshButton: JButton
    private lateinit var serverInfoLabel: JLabel

    private val teamProjects = mutableListOf<String>()

    override fun buildComponent(): JComponent {
        val builder = FormBuilder.createFormBuilder()

        // 服务器信息
        serverInfoLabel = JLabel(" ")
        updateServerInfo()
        builder.addLabeledComponent(TfsBundle.message("wizard.label.connectedTo"), serverInfoLabel)
        builder.addSeparator()
        builder.addLabeledComponent(TfsBundle.message("wizard.label.teamProject"), createTeamProjectPanel())
        builder.addSeparator()
        builder.addComponent(createStatusPanel())

        val panel = builder.panel
        panel.border = JBUI.Borders.empty(10)

        return panel
    }

    /**
     * 更新服务器信息显示
     */
    private fun updateServerInfo() {
        val info = buildString {
            append("<html><b>")
            append(context.serverUrl)
            val collection = context.collectionName
            if (!collection.isNullOrEmpty()) {
                append("/").append(collection)
            }
            append("</b></html>")
        }
        if (::serverInfoLabel.isInitialized) {
            serverInfoLabel.text = info
        }
    }

    /**
     * 创建团队项目选择面板
     */
    private fun createTeamProjectPanel(): JComponent {
        val panel = JPanel(BorderLayout(5, 0))

        teamProjectComboBox = JComboBox()
        teamProjectComboBox.isEnabled = false
        updateTeamProjectComboBox()
        panel.add(teamProjectComboBox, BorderLayout.CENTER)

        val buttonPanel = JPanel(FlowLayout(FlowLayout.LEFT, 5, 0))

        refreshButton = JButton(TfsBundle.message("wizard.button.refresh"))
        refreshButton.addActionListener { loadTeamProjects() }
        buttonPanel.add(refreshButton)

        panel.add(buttonPanel, BorderLayout.EAST)

        return panel
    }

    /**
     * 创建状态面板
     */
    private fun createStatusPanel(): JComponent {
        val panel = JPanel(BorderLayout())
        statusLabel = JBLabel(TfsBundle.message("wizard.status.clickRefresh"))
        panel.add(statusLabel, BorderLayout.CENTER)
        return panel
    }

    /**
     * 加载团队项目列表
     */
    private fun loadTeamProjects() {
        if (!context.isServerConfigured()) {
            statusLabel.text = TfsBundle.message("wizard.error.configureServerFirst")
            statusLabel.foreground = Color.RED
            return
        }

        statusLabel.text = TfsBundle.message("wizard.status.loadingTeamProjects")
        statusLabel.foreground = Color.BLACK
        refreshButton.isEnabled = false
        teamProjectComboBox.isEnabled = false

        ProgressManager.getInstance().run(object : Task.Backgroundable(null, TfsBundle.message("wizard.progress.loadingTeamProjects"), true) {
            override fun run(indicator: ProgressIndicator) {
                indicator.text = TfsBundle.message("wizard.progress.connectingToTfs")
                indicator.isIndeterminate = true

                try {
                    val connectionService = ApplicationManager.getApplication().getService(TfsConnectionService::class.java)
                    val projects = connectionService.getTeamProjects(context)

                    SwingUtilities.invokeLater {
                        teamProjects.clear()
                        teamProjects.addAll(projects)
                        updateTeamProjectComboBox()

                        if (projects.isEmpty()) {
                            statusLabel.text = TfsBundle.message("wizard.status.noTeamProjects")
                            statusLabel.foreground = Color.ORANGE
                        } else {
                            statusLabel.text = TfsBundle.message("wizard.status.teamProjectsLoaded", projects.size)
                            statusLabel.foreground = Color(0, 128, 0)
                        }

                        refreshButton.isEnabled = true
                        teamProjectComboBox.isEnabled = true
                    }

                } catch (e: Exception) {
                    SwingUtilities.invokeLater {
                        statusLabel.text = TfsBundle.message("wizard.error.loadingFailed", e.message ?: "")
                        statusLabel.foreground = Color.RED
                        refreshButton.isEnabled = true
                    }
                }
            }
        })
    }

    /**
     * 更新团队项目下拉框
     */
    private fun updateTeamProjectComboBox() {
        teamProjectComboBox.model = CollectionComboBoxModel(teamProjects)
        if (teamProjects.isNotEmpty()) {
            teamProjectComboBox.selectedIndex = 0
        }
    }

    override fun isComplete(): Boolean = teamProjectComboBox.selectedItem != null

    override fun getPreferredFocusedComponent(): JComponent = teamProjectComboBox

    override fun onFinish(): Boolean {
        val selected = teamProjectComboBox.selectedItem as? String
        if (selected != null) {
            context.teamProject = selected
        }
        return true
    }
}
